pallets/payments/src/lib.rs

```diff
@@ -154,79 +154,61 @@ pub mod pallet {
 	/// interest. The storage map keys are the creator and the recipient, this
 	/// also ensures that for any (sender,recipient) combo, only a single
 	/// payment is active. The history of payment is not stored.
-	pub type Payment<T: Config> = StorageNMap<
+	pub type Payment<T: Config> = StorageDoubleMap<
 		_,
-		(
-			NMapKey<Blake2_128Concat, T::AccountId>,
-			NMapKey<Blake2_128Concat, T::AccountId>,
-			NMapKey<Blake2_128Concat, T::PaymentId>,
-		),
+		Blake2_128Concat,
+		// Sender
+		T::AccountId,
+		Twox64Concat,
+		T::PaymentId,
 		PaymentDetail<T>,
 		ResultQuery<Error<T>::NonExistentStorageValue>,
 	>;
 
 	#[pallet::storage]
-	#[pallet::getter(fn last_id)]
-	pub type LastId<T: Config> = StorageValue<_, T::PaymentId, OptionQuery>;
+	pub type PaymentParties<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		T::PaymentId,
+		// Sender, Beneficiary pair
+		(T::AccountId, T::AccountId),
+		ResultQuery<Error<T>::NonExistentStorageValue>,
+	>;
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// A new payment has been created
 		PaymentCreated {
-			sender: T::AccountId,
-			beneficiary: T::AccountId,
+			payment_id: T::PaymentId,
```

Comment:
just double thinking if its still not useful to display the sender and beneficiaries? i understand the frontends would have these values anyway.